ruby_version_file_parser.go

```diff
@@ -0,0 +1,34 @@
+package bundleinstall
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+)
+
+// RubyVersionFileParser parses the .ruby-version file to determine the
+// version of Ruby used by the application.
+type RubyVersionFileParser struct{}
+
+// NewGemfileParser initializes an instance of RubyVersionFileParser.
+func NewRubyVersionFileParser() RubyVersionFileParser {
+	return RubyVersionFileParser{}
+}
+
+// ParseVersion scans the .ruby-version file for a Ruby version specification.
+func (p RubyVersionFileParser) ParseVersion(path string) (string, error) {
+	rVersion, err := os.ReadFile(path)
+	if err != nil {
+		return "", fmt.Errorf("failed to read .ruby-version file: %w", err)
+	}
+
+	re := regexp.MustCompile(versionNumberExpression)
+	rubyVersion := re.FindString(strings.TrimSpace(string(rVersion)))
+
+	if len(rubyVersion) == 0 {
+		return "", fmt.Errorf("no valid ruby version found in .ruby-version file: %s", rVersion)
```

Comment:
I think I agree that this should return an error, as if the `.ruby-version` file exists but does not contain a valid ruby version that's essentially a syntactical error in the application and the buildpack should bail out.

However, i notice that the Gemfile parser _does not_ return an error in the case where the Gemfile doesn't contain a ruby version. But I think that's fine because it's not a required part of that file.

@ctrox @sophiewigmore do you agree with my logic above?

Reply:
I think it makes sense for the parser to error out like it's doing here, and then if it errors, during detection we just emit a warning based on the parser error. https://github.com/paketo-buildpacks/bundle-install/pull/499#discussion_r1395778416

I think it makes sense that the build would try to proceed without a .ruby-version if it can